bin/www/localPackage.js

```diff
@@ -91,11 +91,16 @@ var LocalPackage = (function (_super) {
                     else {
                         var invokeSuccessAndInstall = function () {
                             CodePushUtil.logMessage("Install succeeded.");
-                            installSuccess && installSuccess();
-                            cordova.exec(function () { }, function () { }, "CodePush", "install", [deployDir.fullPath, installOptions.rollbackTimeout.toString(), installOptions.installMode.toString()]);
+                            if (installOptions.installMode === InstallMode.IMMEDIATE) {
+                                installSuccess && installSuccess();
+                                cordova.exec(function () { }, function () { }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            }
+                            else {
+                                cordova.exec(function () { installSuccess && installSuccess(); }, function () { installError && installError(); }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            }
                         };
                         var preInstallSuccess = function () {
-                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded);
+                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded, _this);
```

Comment:
`reportStatus()` is now used to record that a user has either 1. upgraded successfully to a new version or 2. failed to upgrade to a new version (due to a crash triggering a rollback, or an install failure). Therefore, I think we should report DeploymentSucceeded during `notifyApplicationReady()` if we see that the currentPackage has isFirstRun set to true, because that is the only time we can ascertain that an app has been installed successfully.

Also, we will need a new mechanism to find out that an app has been rolled back, and call reportStatus() there.